packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -208,6 +208,13 @@ export function updateWrapper(element: Element, props: Object) {
 export function postMountWrapper(element: Element, props: Object) {
   const node = ((element: any): InputWithWrapperState);
 
+  // Submit or reset inputs should not have their value overridden
+  // with a falsy value, or else this will result in a button with no text,
+  // ignoring the browser-default "Submit", etc.
+  if (!props.value && (props.type === 'submit' || props.type === 'reset')) {
```

Comment:
This would make it impossible to intentionally set an empty value

```jsx
<input type="submit" value="" />
```

We'd need a more specific check like `props.value === undefined`. It should probably be moved into the `hasOwnProperty` check below, so we can avoid checking `type` if `value` doesn't exist at all.